instrumentation/runtime/runtime.go

```diff
@@ -20,48 +20,100 @@ import (
+// WithMinimumReadMemStatsInterval() option to modify this setting in
 // Start().
-const DefaultMinimumGCStatsInterval time.Duration = 15 * time.Second
+const DefaultMinimumReadMemStatsInterval time.Duration = 15 * time.Second
 
-// WithMinimumGCStatsInterval sets a minimum interval between calls to
+// WithMinimumReadMemStatsInterval sets a minimum interval between calls to
 // runtime.ReadMemStats(), which is a relatively expensive call to make
-// frequently.  `d` values less than 0 will be disregarded silently.
-func WithMinimumGCStatsInterval(d time.Duration) Option {
-	return func(r *runtime) {
-		if d >= 0 {
-			r.minGCStatsInterval = d
-		}
+// frequently.  This setting is ignored when `d` is negative.
+func WithMinimumReadMemStatsInterval(d time.Duration) Option {
+	return minimumReadMemStatsIntervalOption(d)
+}
+
+type minimumReadMemStatsIntervalOption time.Duration
+
+// ApplyRuntime implements Option.
+func (o minimumReadMemStatsIntervalOption) ApplyRuntime(c *Config) {
+	if o >= 0 {
+		c.MinimumReadMemStatsInterval = time.Duration(o)
 	}
 }
 
-// New returns Runtime, a structure for reporting Go runtime metrics
-// interval is used to limit how often to invoke Go runtime.ReadMemStats() to obtain metric data.
-// If the metric SDK attempts to observe MemStats-derived instruments more frequently than the
-// interval, a cached value will be used.
-func Start(provider metric.Provider, opts ...Option) error {
-	r := &runtime{
-		// TODO: How to set the instrumentation library version?
-		meter:              provider.Meter("runtime"),
-		minGCStatsInterval: DefaultMinimumGCStatsInterval,
+// WithMeterProvider sets the Metric implementation to use for
+// reporting.  If this option is not used, the global metric.Provider
+// will be used.  `provider` must be non-nil.
+func WithMeterProvider(provider metric.Provider) Option {
+	return metricProviderOption{provider}
+}
+
+type metricProviderOption struct{ metric.Provider }
+
+// ApplyRuntime implements Option.
+func (o metricProviderOption) ApplyRuntime(c *Config) {
+	c.MeterProvider = o.Provider
+}
+
+// Configure computes a Config from the supplied Options.
+func Configure(opts ...Option) Config {
+	c := Config{
+		MeterProvider:               global.MeterProvider(),
+		MinimumReadMemStatsInterval: DefaultMinimumReadMemStatsInterval,
 	}
 	for _, opt := range opts {
-		opt(r)
+		opt.ApplyRuntime(&c)
+	}
+	return c
+}
+
+// Start initializes reporting of runtime metrics using the supplied Config.
+func Start(c Config) error {
+	if c.MinimumReadMemStatsInterval < 0 {
+		c.MinimumReadMemStatsInterval = DefaultMinimumReadMemStatsInterval
+	}
+	if c.MeterProvider == nil {
+		c.MeterProvider = global.MeterProvider()
+	}
+	r := &runtime{
+		meter: c.MeterProvider.Meter(
+			// TODO: should library names be qualified?
+			// e.g., contrib/runtime?
+			"runtime",
+			// TODO(#225): set the instrumentation library version
+			// metric.WithInstrumentationVersion(contrib.SemVersion())
```

Comment:
#225 is merged to mainline, so should be able to uncomment this now.

Reply:
Done.